content/influxdb/cloud-dedicated/query-data/influxql/parameterized-queries.md

```diff
@@ -0,0 +1,304 @@
+
+- Null
+- Boolean
+- Unsigned integer (`u_int64`)
+- Integer (`int64`)
+- Double (`float64`)
+- String
+
+### Time expressions
+
+To parameterize time bounds, substitute a parameter for a timestamp literal--for example:
+
+```sql
+SELECT *
+FROM home
+WHERE time >= $min_time`
+```
+
+For the parameter value, specify the timestamp literal as a string--for example:
+
+{{% influxdb/custom-timestamps %}}
+
+```go
+// Assign a timestamp string literal to the min_time parameter.
+parameters := influxdb3.QueryParameters{
+    "min_time": "2024-03-18 00:00:00.00",
+}
+```
+
+{{% /influxdb/custom-timestamps %}}
+
+### Not supported
+
+Some data types don't support parameters--for example, in InfluxQL queries, InfluxDB doesn't support parameter substitution for durations or inside of duration values--for example, the following won’t work:
+
+  ```go
+  query := `
+      SELECT avg(temp) as temp
+      FROM home
+      WHERE time >= now() - $days
+      GROUP BY time(2h), room`
+
+  parameters := influxdb3.QueryParameters{
+      "days": "7d",
+  }
+  ```
+
+## Parameterize an InfluxQL query
+
+{{% note %}}
+#### Sample data
+
+The following examples use the
+[Get started home sensor data](/influxdb/cloud-dedicated/reference/sample-data/#get-started-home-sensor-data).
+To run the example queries and return results,
+[write the sample data](/influxdb/cloud-dedicated/reference/sample-data/#write-the-home-sensor-data-to-influxdb)
+to your {{% product-name %}} database before running the example queries.
+{{% /note %}}
+
+To use a parameterized query, do the following:
+
+1. In your query text, use the `$parameter` syntax to reference a parameter name--for example,
+the following query contains `$room` and `$min_temp` parameter placeholders:
+
+   ```sql
+   SELECT *
+   FROM home
+   WHERE time > now() - 7d
+   AND temp >= $min_temp
+   AND room = $room
+   ```
+
+2. Provide a value for each parameter name.
+   If you don't assign a value for a parameter, InfluxDB returns an error.
+   The syntax for providing parameter values depends on the client you use--for example:
+
+   <!-- I expect to add more client examples soon -->
+
+   {{< code-tabs-wrapper >}}
+   {{% code-tabs %}}
+   [Go](#)
+   {{% /code-tabs %}}
+   {{% code-tab-content %}}
+   <!------------------------ BEGIN GO ------------------------------------------->
+   ```go
+   // Define a QueryParameters struct--a map of parameters to input values.
+   parameters := influxdb3.QueryParameters{
+       "room": "Kitchen",
+       "min_temp": 20.0,
+   }
+   ```
+   <!-------------------------- END GO ------------------------------------------->
+   {{% /code-tab-content %}}
+   {{< /code-tabs-wrapper >}}
+
+After InfluxDB receives your request and parses the query, it executes the query as
+
+```sql
+SELECT *
+FROM home
+WHERE time > now() - 7d
+AND temp >= 20.0
+AND room = 'Kitchen'
```

Comment:
Question--if the parameter is passed as a string, does this actually get executed as a string literal (with single quotes `''`)? And if so, how do you parameterize an identifier (either no quotes or double quotes `""`). For example:

```sql
SELECT *
FROM $table
WHERE time > now() - 7d
AND temp >= $min_temp
AND room = $room
```

With the following parameter definition:

```go
   parameters := influxdb3.QueryParameters{
       "table": "home",
       "room": "Kitchen",
       "min_temp": 20.0,
   }
```

I would expect `$table` to be executed as an identifier, not a string literal.

```sql
SELECT *
FROM "home"
WHERE time > now() - 7d
AND temp >= 20.0
AND room = 'Kitchen'
```

So... when building a parameterized query, does the user need to include double- vs single-quotes in the template or is the engine smart enough to figure it out. I would almost expect to have to include the quotes in the template to make sure it works correctly:

```sql
SELECT *
FROM "$table"
WHERE time > now() - 7d
AND temp >= $min_temp
AND room = '$room'
```

Reply:
As far as I can tell, you can't use parameters for identifiers or literals yet, nor as expressions in `SELECT` or function args.  That's why I was specific about this:

> you can use parameters in `WHERE` clause **predicate expressions**

I tried several permutations for identifiers and literals and couldn't make them work, except for timestamp literals. And, since `literal` isn't in `ReadInfo params`, I assume it's not supported.

I didn't verify everything yet with @erratic-pattern, but I intended to go back and add more.

Reply:
Ah, I see it now. I think we should probably call this limitation/feature out sooner in the doc, possibly even as a note. Right now, the only place I see it is under "Parameter data types."